### PR TITLE
webui: new item filter

### PR DESCRIPTION
**Pull request by or2e**
1. Added "Show only new items" filter
   The item is considered new if it was added not earlier than **30** days
   Value is changed in `item-service`, I did not move it in `constants.json`
    <details>
    <summary>Screenshot</summary>
    
    image
    </details>
2. Changed the dropdown behavior of additional filters. 
    Now it closes after selecting
    Also made the open button more visible
    <details>
    <summary>Demo</summary>
   

    https://github.com/verdie-g/crpg/assets/33551334/7864aed4-54fb-4f8e-ab05-072a739c5af2
    </details>
3. Fixed auto-tests related to search service and indexer

### CODE REVIEW

File: src/WebUI/src/services/item-search-service/indexator.ts [hunk 1]
```diff
@@ -185,6 +185,10 @@ const mapMountProps = (item: Item) => {
   };
 };
 
+const isNewItemDays = 30;
+// TODO: const name
+const prior = new Date().setDate(new Date().getDate() - isNewItemDays);
```
Comment (verdie-g) [inline src/WebUI/src/services/item-search-service/indexator.ts:190]: newItemDateThreshold 🤷

File: src/WebUI/src/pages/shop.vue [hunk 2]
```diff
@@ -1,6 +1,11 @@
 <script setup lang="ts">
-import { WeaponUsage, type ItemFlat, ItemCompareMode } from '@/models/item';
```
Comment (or2e) [inline src/WebUI/src/pages/shop.vue:2]: `ItemCompareMode` - unused

File: src/WebUI/locales/en.yml [hunk 3]
```diff
@@ -1101,6 +1101,10 @@ item:
       title: 'Rank'
       description: ''
 
+    new:
+      title: 'New items only'
+      description: 'Show items added within {days} days'
```
Comment (verdie-g) [inline src/WebUI/locales/en.yml:1106]:
```suggestion
      description: 'Show items added or reworked within the last {days} days'
```
Same for CN/RU

File: src/WebUI/src/services/item-search-service/indexator.ts [hunk 4]
```diff
@@ -195,12 +197,13 @@ const itemToFlat = (item: Item): ItemFlat => {
   ];
 
   // Banning the use of large shields on horseback
-  if (isLargeShield(item)) {
+  if (item.type === ItemType.Shield && isLargeShield(item)) {
```
Comment (verdie-g) [inline src/WebUI/src/services/item-search-service/indexator.ts:200]: That check is already done in isLargeShield

**Review by verdie-g — CHANGES_REQUESTED** [review]

**Review by verdie-g — APPROVED** [review]

**Pull request merged**